opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,722 @@
+
+        train_examples = self.load_sentence_transformer_example(
+            query_df, use_accelerate
+        )
+
+        if use_accelerate is True:
+            self.set_up_accelerate_config(
+                compute_environment=compute_environment,
+                num_machines=num_machines,
+                num_processes=num_processes,
+            )
+
+            if self.__is_notebook():
+                notebook_launcher(
+                    self.train_model,
+                    args=(
+                        train_examples,
+                        self.model_id,
+                        output_model_path,
+                        output_model_name,
+                        use_accelerate,
+                        learning_rate,
+                        num_epochs,
+                        verbose,
+                    ),
+                    num_processes=num_processes,
+                )
+            else:
+                subprocess.run(
+                    [
+                        "accelerate launch self.train_model(train_examples, self.model_id, output_model_path, output_model_name,use_accelerate,learning_rate,num_epochs,verbose)"
+                    ]
+                )
+        else:
+            self.train_model(
+                train_examples,
+                self.model_id,
+                output_model_path,
+                output_model_name,
+                use_accelerate,
+                learning_rate,
+                num_epochs,
+                verbose,
+            )
+
+        self.zip_model(output_model_path, zip_file_name)
+        return None
+
+    #    public step by step functions:
+    def read_queries(self, read_path: str) -> pd.DataFrame:
+        """
+        Description:
+        Read the queries generated from the Synthetic Query Generator (SQG) model
+
+        Parameters:
+        read_path: str
+            required, path to the zipped file that contains generated queries, if None, raise exception
+
+        Return:
+            The dataframe of queries.
+        """
+
+        process = []
+        file_list = []
+
+        if read_path is None:
+            raise Exception(
+                "No file provided. Please provide the path to synthetic query zip file."
+            )
+
+        # assign a local folder 'synthetic_queries/' to store the unzip file,
+        # check if the folder contains sub-folders and files, remove and clean up the folder before unzip.
+        # walk through the zip file and read the file paths into file_list
+        unzip_path = os.path.join(os.getcwd(), "synthetic_queries/")
+
+        # ML add warning here and confirm with user to proceed
+        if os.path.exists(unzip_path):
+            if len(os.listdir(unzip_path)) > 0:
+                for files in os.listdir(unzip_path):
+                    sub_path = os.path.join(unzip_path, files)
+                    if os.path.isfile(sub_path):
+                        os.remove(sub_path)
```

Comment:
Delete all files under a path is risky. How about we throw error if the unzip_path is not empty?

Reply:
added in the commit [add overwrite to clean up synthetic_queries folder](https://github.com/opensearch-project/opensearch-py-ml/pull/18/commits/f54b9758b5f6517fd32848d0612b4ed9066289fb)